### ISSUE TITLE
Api Node crash with out of order statements.

### ISSUE THREAD

**Issue by Wayonb** [nemtech/catapult-server#148]
````
2021-01-26 15:58:04.350914 0x00007f5a4a238700: <debug> (disruptor::Disruptor.cpp@43) disruptor queuing element 13127120 (3 txes [00000000] from Remote_Push with size 3KB 111B) 
2021-01-26 15:58:04.352291 0x00007f5a4a238700: <debug> (disruptor::Disruptor.cpp@43) disruptor queuing element 13127130 (3 txes [00000000] from Remote_Push with size 3KB 14B) 
2021-01-26 15:58:04.352521 0x00007f5a4a238700: <debug> (disruptor::Disruptor.cpp@43) disruptor queuing element 13127140 (4 txes [00000000] from Remote_Push with size 4KB 146B) 
2021-01-26 15:58:04.392880 0x00007f59e57a4700: <info> (disruptor::ConsumerDispatcher.cpp@44) completing processing of element 13127120 (3 txes [DD3C318C] empty from Remote_Push with size 3KB 111B), last consumer is 21 elements behind 
2021-01-26 15:58:04.392987 0x00007f59e57a4700: <info> (disruptor::ConsumerDispatcher.cpp@44) completing processing of element 13127130 (3 txes [A7AAA2CC] from Remote_Push with size 3KB 14B), last consumer is 11 elements behind 
2021-01-26 15:58:04.393055 0x00007f59e57a4700: <info> (disruptor::ConsumerDispatcher.cpp@44) completing processing of element 13127140 (4 txes [B488F7EB] from Remote_Push with size 4KB 146B), last consumer is 1 elements behind 
2021-01-26 15:58:04.420163 0x00007f5a4a238700: <debug> (utils::StackLogger.h@37) pushing scope 'generating candidate block' 
2021-01-26 15:58:04.471022 0x00007f5a4aa39700: <debug> (chain::EntitiesSynchronizer.h@51) peer returned 4 unconfirmed transactions 
2021-01-26 15:58:04.471328 0x00007f5a4aa39700: <info> (chain::RemoteApiForwarder.h@68) completed 'pull unconfirmed transactions task' (mikun_2 @ 80.208.229.251:7900) with result Success 
2021-01-26 15:58:04.543932 0x00007f5a4a238700: <error> (model::ResolutionStatement.cpp@77) Throwing exception: detected out of order resolution - last (2, 1) next (2, 0) 
2021-01-26 15:58:04.604028 0x00007f5a4a238700: <fatal> (process::ProcessMain.cpp@75) 
thread: 1 (server) work
unhandled exception while running local node!
/catapult-src/src/catapult/model/ResolutionStatement.cpp(77): Throw in function void catapult::model::ResolutionStatement<TUnresolved, TResolved, ResolutionReceiptType>::addResolution(const TResolved&, const catapult::model::ReceiptSource&) [with TUnresolved = catapult::utils::BaseValue<long unsigned int, catapult::UnresolvedMosaicId_tag>; TResolved = catapult::utils::BaseValue<long unsigned int, catapult::MosaicId_tag>; catapult::model::ReceiptType ResolutionReceiptType = (catapult::model::ReceiptType)62019]
Dynamic exception type: boost::wrapexcept<catapult::catapult_error<std::invalid_argument> >
std::exception::what: detected out of order resolution - last (2, 1) next (2, 0)
````

**Comment by meyns** [comment]
Same error here:
2021-01-26 15:59:56.364728 0x00007fe7b1e34700: <error> (model::ResolutionStatement.cpp@77) Throwing exception: detected out of order resolution - last (2, 1) next (2, 0)
2021-01-26 15:59:56.511900 0x00007fe7b1e34700: <fatal> (process::ProcessMain.cpp@75)
thread: 2 (server) work
unhandled exception while running local node!
/catapult-src/src/catapult/model/ResolutionStatement.cpp(77): Throw in function void catapult::model::ResolutionStatement<TUnresolved, TResolved, ResolutionReceiptType>::addResolution(const TResolved&, const catapult::model::ReceiptSour>
Dynamic exception type: boost::wrapexcept<catapult::catapult_error<std::invalid_argument> >
std::exception::what: detected out of order resolution - last (2, 1) next (2, 0)

**Comment by Jaguar0625** [comment]
Fixed in 510dba8ec.
